### PR TITLE
Injection caveats docs

### LINKED ISSUE #4253: Docs should include additional caveats around sidecar injection
Docs for sidecar injection should mention the following two caveats:
* Sidecar injection seems to be impossible in `kube-system` and `public` namespaces, as per the declaration of `ignoredNamespaces` [here](https://github.com/istio/istio/blob/13cb3422ae47f95dfbc829a327724a56929cab3b/pilot/pkg/kube/inject/initializer.go#L28) and the call to `injectRequired` [here](https://github.com/istio/istio/blob/13cb3422ae47f95dfbc829a327724a56929cab3b/pilot/pkg/kube/inject/webhook.go#L558).
* Sidecar injection doesn't work for pods on the host network, as per code [here](https://github.com/istio/istio/blob/13cb3422ae47f95dfbc829a327724a56929cab3b/pilot/pkg/kube/inject/inject.go#L403)

Both these facts seem to be undocumented. Happy to provide a PR tomorrow.

### PR DESCRIPTION
**Pull request by Niksko**
Closes #4253

### CODE REVIEW

File: content/docs/ops/setup/injection/index.md [hunk 1]
```diff
@@ -40,6 +45,16 @@ these three variables.
 This includes an injected sidecar when it wasn't expected and a lack
 of injected sidecar when it was.
 
+1. Ensure your pod is not in the `kube-system` or `kube-public` namespace.
+   Automatic sidecar injection cannot be used in these namespaces.
+
+1. Ensure your pod does not have `hostNetwork: true` in its pod spec.
+   Automatic sidecar injection cannot be used with pods that are on the host network.
+
+   The sidecar model assumes that the iptables changes required for Envy to intercept
```
Comment (frankbu) [inline content/docs/ops/setup/injection/index.md]: Envoy

File: content/docs/ops/setup/injection/index.md [hunk 2]
```diff
@@ -11,14 +11,19 @@ time. Injection can be scoped to particular sets of namespaces using
 the webhooks `namespaceSelector` mechanism. Injection can also be
 enabled and disabled per-pod with an annotation.
 
-Whether or not a sidecar is injected is depends on three things:
+Whether or not a sidecar is injected depends on three pieces of configuration and two limitations:
```
Comment (linsun) [inline content/docs/ops/setup/injection/index.md]: I would not position is as 2 limitations... it is something by design and don't want to confuse users.

File: content/docs/ops/setup/injection/index.md [hunk 3]
```diff
@@ -11,14 +11,19 @@ time. Injection can be scoped to particular sets of namespaces using
...
 
+Configuration:
 * webhooks `namespaceSelector`
 * default `policy`
 * per-pod override annotation
 
+Limitations:
+* sidecars cannot be injected in the `kube-system` or `kube-public` namespaces
```
Comment (linsun) [inline content/docs/ops/setup/injection/index.md]: what about istio-system namespace?

File: content/docs/ops/setup/injection/index.md [hunk 4]
```diff
@@ -40,6 +45,16 @@ these three variables.
 This includes an injected sidecar when it wasn't expected and a lack
 of injected sidecar when it was.
 
+1. Ensure your pod is not in the `kube-system` or `kube-public` namespace.
+   Automatic sidecar injection cannot be used in these namespaces.
```
Comment (frankbu) [inline content/docs/ops/setup/injection/index.md]: Can you be more clear about the limitation, e.g., what would happen if you turn on an annotation? Does Istio ignore it, or will very bad things happen :-) Saying something like `Automatic sidecar injection configuration is ignored in these namespaces.` would be less mysterious IMO.

File: content/docs/ops/setup/injection/index.md (hunk 2 above)
Comment (sdake) [reply content/docs/ops/setup/injection/index.md]: instead of two limitations, you might try `two security rules`.

**Comment by tangiel** [comment]
@googlebot rescan

File: content/docs/ops/setup/injection/index.md (hunk 3 above)
Comment (Niksko) [reply content/docs/ops/setup/injection/index.md]: My reading of the code didn't show any limitations on `istio-system`, but happy to add it

File: content/docs/ops/setup/injection/index.md [hunk 5]
```diff
@@ -11,14 +11,19 @@ time. Injection can be scoped to particular sets of namespaces using
 the webhooks `namespaceSelector` mechanism. Injection can also be
 enabled and disabled per-pod with an annotation.
 
-Whether or not a sidecar is injected is depends on three things:
+Whether or not a sidecar is injected depends on three pieces of configuration and two security rules:
 
+Configuration:
```
Comment (frankbu) [inline content/docs/ops/setup/injection/index.md]: These lists are displaying correctly in the generated preview: https://deploy-preview-4255--preliminary-istio.netlify.com/

I think Hugo only accepts `-` (not `*`) for list items, so it needs to be:
```
Configuration:
- webhooks `namespaceSelector`
- default `policy`
- per-pod override annotation
```

Comment (Niksko) [reply content/docs/ops/setup/injection/index.md]: Turns out it needs whitespace around it

**Pull request merged**